Build fails with Go 1.26 rc1

Build problem found downstream in 
* https://github.com/Homebrew/homebrew-core/pull/258912

Confirmed locally via simple `go build` command inside this repository:
```
$ go1.26rc1 build ./...
# vitess.io/vitess/go/hack
go/hack/ensure_swiss_map.go:39:38: invalid array length -1 (untyped int constant)
```

The Swiss-maps-ensuring Go build flag:
https://github.com/vitessio/vitess/blob/b3666b2d0e3c1809d95c906b0b15e4f7499ba951/go/hack/ensure_swiss_map.go#L1
is no longer good enough for Go 1.26, as in Go 1.26 and later Swiss maps are enabled by default and there is no more `GOEXPERIMENT=noswissmap` nor `GOEXPERIMENT=swissmap` flag either. 

See:
* https://github.com/golang/go/commit/2ae059ccaf982c3304fae0b48c1d78ad7192cbdd
* https://github.com/golang/go/issues/54766

--- a/go/hack/ensure_swiss_map.go
+++ b/go/hack/ensure_swiss_map.go
@@ -1,4 +1,4 @@
-//go:build !goexperiment.swissmap
+//go:build !goexperiment.swissmap && !go1.26
 
 /*
 Copyright (c) 2009 The Go Authors. All rights reserved.
@@ -32,7 +32,8 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 // This is invalid Go code, and it will fail to compile if you disable
 // Swiss maps when building Vitess. Our runtime memory accounting system
-// expects the map implementation in Go 1.24 to be Swiss.
+// expects the map implementation in Go 1.24 and 1.25 to be Swiss.
+// In Go 1.26 and later, Swiss maps are always enabled.
 
 package hack
 

